core/exception_vectors.S

```diff
@@ -81,7 +83,9 @@ KernelExceptionVector:
         .type   KernelExceptionVector, @function
 
         break   1, 0
-        call0   sdk_user_fatal_exception_handler
+        mov     a2, a1
+        movi    a3, 0
```

Comment:
The register state is only dumped if the second parameter (a3) is set to 1, and it dumps the register as they were saved onto the stack by _xt_int_enter/xtensa_context_save.

I think we could modify the exception handlers to always have access to registers in a fatal exception, but this is less intrusive (and most exceptions happen via the user exception path.) Could be expanded in the future, though.